geode-assembly/src/acceptanceTest/java/org/apache/geode/launchers/LocatorLauncherWithCustomLogConfigAcceptanceTest.java

```diff
@@ -0,0 +1,175 @@
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.launchers;
+
+import static org.apache.geode.test.awaitility.GeodeAwaitility.await;
+import static org.apache.geode.test.util.ResourceUtils.createFileFromResource;
+import static org.apache.geode.test.util.ResourceUtils.getResource;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import org.apache.geode.internal.AvailablePortHelper;
+import org.apache.geode.test.assertj.LogFileAssert;
+import org.apache.geode.test.junit.rules.RequiresGeodeHome;
+
+public class LocatorLauncherWithCustomLogConfigAcceptanceTest {
+
+  private static final String CONFIG_FILE_NAME =
+      "LocatorLauncherWithCustomLogConfigAcceptanceTest.xml";
+  private static final String LOCATOR_NAME = "the-locator";
+
+  private int locatorPort;
+  private Path configFile;
+  private Process locator;
+  private Path geodeDependencies;
+  private Path stdoutFile;
+  private Path locatorLogFile;
+  private Path pulseLogFile;
+
+  @Rule
+  public RequiresGeodeHome requiresGeodeHome = new RequiresGeodeHome();
+  @Rule
+  public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  @Before
+  public void setUpLogConfigFile() {
+    configFile = createFileFromResource(getResource(CONFIG_FILE_NAME), temporaryFolder.getRoot(),
+        CONFIG_FILE_NAME)
+            .toPath();
+  }
+
+  @Before
+  public void setUpGeodeDependencies() {
+    Path geodeHome = requiresGeodeHome.getGeodeHome().toPath();
+    geodeDependencies = geodeHome.resolve("lib/geode-dependencies.jar");
+
+    assertThat(geodeDependencies).exists();
+  }
+
+  @Before
+  public void setUpOutputFiles() {
+    stdoutFile = temporaryFolder.getRoot().toPath().resolve("stdout.txt");
+    locatorLogFile = temporaryFolder.getRoot().toPath().resolve(LOCATOR_NAME + ".log");
+    pulseLogFile = temporaryFolder.getRoot().toPath().resolve("pulse.log");
+  }
+
+  @Before
+  public void setUpRandomPorts() {
+    locatorPort = AvailablePortHelper.getRandomAvailableTCPPort();
+  }
+
+  @After
+  public void stopLocator() throws Exception {
+    if (locator != null) {
+      locator.destroyForcibly().waitFor(4, TimeUnit.SECONDS);
+    }
+  }
+
+  @Test
+  public void locatorLauncherUsesLog4jConfigurationFile() throws Exception {
+    ProcessBuilder processBuilder = new ProcessBuilder()
+        .redirectErrorStream(true)
+        .redirectOutput(stdoutFile.toFile())
+        .directory(temporaryFolder.getRoot())
+        .command("java",
```

Comment:
If system property java.home is defined, please use `System.getProperty("java.home")+"/bin/java"` instead.  Only use `java` if system property java.home is not defined

Reply:
Thanks Owen!